src/group_theory/perm/cycle_type.lean

```diff
@@ -39,75 +39,80 @@ variables [decidable_eq α]
 
 /-- The cycle type of a permutation -/
 def cycle_type (σ : perm α) : multiset ℕ :=
-σ.trunc_cycle_factors.lift (λ l, l.1.map (finset.card ∘ support))
-  (λ ⟨l₁, h₁l₁, h₂l₁, h₃l₁⟩ ⟨l₂, h₁l₂, h₂l₂, h₃l₂⟩, coe_eq_coe.mpr (perm.map _
-  (list_cycles_perm_list_cycles (h₁l₁.trans h₁l₂.symm) h₂l₁ h₂l₂ h₃l₁ h₃l₂)))
+σ.cycle_factors_finset.1.map (finset.card ∘ support)
```

Comment:
I don't think I ever saw the map `finset α → multiset α` expressed as a projection outside of the most basic files, but it does seem to be the canonical way to write this.